### PR TITLE
Fix resource leaks in Input Actions Editor window.

### PR DESCRIPTION
**Pull request by AlexTyrer**
### Description

This PR addresses resource leaks in the Input Actions Editor window.

These resource leaks could lead to unexpected exceptions being thrown.

### Changes made

InputActionsEditorSettingsProvider is now handled by a single static instance to work around UIToolkit callback order issues identified.

Fix VisualElement handling on the StateContainer that manages the tracked serialized object representing the Input Actions.

### Checklist

Before review:

- [x] Changelog entry added.
    - Explains the change in `Changed`, `Fixed`, `Added` sections.
    - For API change contains an example snippet and/or migration example.
    - FogBugz ticket attached, example `([case %number%](https://issuetracker.unity3d.com/issues/...))`.
    - FogBugz is marked as "Resolved" with *next* release version correctly set.
- [x] Tests added/changed, if applicable.
    - Functional tests `Area_CanDoX`, `Area_CanDoX_EvenIfYIsTheCase`, `Area_WhenIDoX_AndYHappens_ThisIsTheResult`.
    - Performance tests.
    - Integration tests.
- [x] Docs for new/changed API's.
    - Xmldoc cross references are set correctly.
    - Added explanation how the API works.
    - Usage code examples added.
    - The manual is updated, if needed.

- [x] Commit message for squash-merge is prefixed with one of the list:
    - FIX: Fixed Input Actions Editor window resource leak that could result in unexpected exceptions [ISXB-865](https://jira.unity3d.com/browse/ISXB-865)

### CODE REVIEW

File: Packages/com.unity.inputsystem/CHANGELOG.md
Comment (ekcoh) [inline Packages/com.unity.inputsystem/CHANGELOG.md]: This should not use internal URL. If internal only provide issue reference number or skip.

File: Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/StateContainer.cs
Comment (ekcoh) [inline Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/StateContainer.cs:51]: Maybe add a comment here explaining why we do this little dance?
E.g. "We need to use a root element for TrackSerializedObjectValue that is destroyed with the view. Using a root element from the settings window would not enable the tracking callback to be destroyed or garbage collected."

**Review by ekcoh — CHANGES_REQUESTED** [review]
Two small comments to address, otherwise LGTM. Will approve once updated if QA is positive.

File: Packages/com.unity.inputsystem/CHANGELOG.md
Comment (AlexTyrer) [reply Packages/com.unity.inputsystem/CHANGELOG.md]: Of course - I've removed the internal link in 5e872116617b56d20a930efac85404e519874a53

File: Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/StateContainer.cs
Comment (AlexTyrer) [reply Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/StateContainer.cs:51]: Good idea - have added the comment in 5e872116617b56d20a930efac85404e519874a53

**Review by ekcoh — APPROVED** [review]

**Comment by ekcoh** [comment]
I think this can land after QA assessment is available with approval.

**Review by Pauliusd01 — APPROVED** [review]
Since I wasn't able to repro the original issues in Unity 6 I only checked basic ProjectWideActions and custom actions window functionality. Did not notice any new issues from these changes. LGTM

**Pull request merged**
